### ISSUE TITLE
[Mac] Bookmark and iCloud sync not working with VPN on

### ISSUE THREAD

**Issue by mikesk8** [mullvad/mullvadvpn-app#2401]
Hello, 

I have noticed that a sync between Safari bookmarks on Mac (10.14.6) and iPhone (14.3) stopped to work some time ago. I did a deep dive and it turned out that switching the VPN (both latest Mullvad apps) off on both devices makes the bookmarks to sync again. 

Have you heard about this before?

Thanks, m

**Comment by faern** [comment]
Thank you for the report. Yes we recently started getting reports about this and have started investigating a little bit. Let's keep this issue as a tracker for when we have it resolved.

**Comment by SukinoSenze** [comment]
I've experienced some kind of issue.

when keeping the VPN connection on, 
`messages` and `photos` apps on my MacBook no longer sync with my iPhone.

**Comment by faern** [comment]
These Apple services probably use some protocol/ip range that we are blocking for security reasons. If we find out what IPs/ports it's using we can consider if unblocking them would be safe or not.

Does this syncing stop working if the app is running on only one of the devices, or does our VPN have to run on both for the syncing to break? If only running it on one of them stops the syncing, which one? Only the mac or only the iPhone?

**Comment by SukinoSenze** [comment]
I'm confused and cannot reproduce it exactly.

the most problem I experienced is
deleting SMS message on iPhone -> not synced to mac MacBook, the message still exists on MacBook.

Mullvad VPN: ON (iPhone + Mac)
-> Not working

Mullvad VPN: OFF (iPhone + Mac)
-> Still not working (I don't know why but it was working before I started using Mullvad first time)

I removed Mullvad and using Cloudflare WARP instead (iPhone + Mac)
-> It's working now, deleting a message is works on both devices.

I think it's hard to reproduce and I'm not sure 100% it's from Mullvad.
But I pretty much sure that it's always working before I use Mullvad the first time.

May be others can give more diagnose detail.

**Comment by steve23094** [comment]
I spent considerable time diagnosing this way back at the beginning of January, I had hoped this information would have been passed onto you.

1. It's the Mullvad macOS app. The iOS app does not block syncing in any way.
2. It's definitely the Mullvad macOS app. The Wireguard app does not block syncing in any way.

**Comment by faern** [comment]
Thanks for the extra info! Yeah I suspected so. Because the macOS app does pretty strict firewalling. I suppose it's using some multicast protocol that we block. We have an internal task to look at iPhone syncing.

**Comment by steve23094** [comment]
Any news on this please? I'd like to go back to using the Mullvad app, but can't considering how important it is to sync this data.

**Comment by steve23094** [comment]
@faern Looking at the changelog for version 2021.3 it doesn't appear that this has been addressed. Has somebody been assigned to this yet? Have they looked at it?

Thanks.

**Comment by faern** [comment]
We have sadly not had the time to investigate this further yet.

**Comment by space-chicken** [comment]
Interesting observation:

I had the exact same problem with Cloudflare's VPN (WARP). But with the new update, they have introduced a new 'local proxy mode'.

Using the local proxy fixes the problem and bookmarks now sync perfectly with iOS devices. Connecting using the standard mode still has the issue.

**Comment by possiblerobot** [comment]
Any word on this issue? The MacOS app is still blocking bookmark syncing for Safari (not sure about iOS). It may be causing issues with iCloud tabs too.

Thanks!

**Comment by dahmc** [comment]
This is the issue that prevents me to switch to mullvad, as it's essential to my workflow.

**Comment by steve10883** [comment]
Another release and still not fixed. Unbelievable.

[https://github.com/mullvad/mullvadvpn-app/releases/tag/2021.6](url)

**Comment by paulrudy** [comment]
I just discovered this thread when searching for a solution to the same problem. I guess I'll have to switch VPN providers. I was hoping there would be a straightforward solution.

**Comment by steve10883** [comment]
> I just discovered this thread when searching for a solution to the same problem. I guess I'll have to switch VPN providers. I was hoping there would be a straightforward solution.

The official Wireguard app works. But don’t ever expect Mullvad to fix their own app.

**Comment by paulrudy** [comment]
Good to know Wireguard works, thanks

**Comment by mikesk8** [comment]
It is a pity Mac users are not so important :( I have been waiting for this feature for a very long time!

**Comment by possiblerobot** [comment]
There seems to be an issue with the syncing of tab groups as well. It could be related to the bookmark syncing issue. As soon as Mullvad is disconnected, bookmarks and tab groups behave normally again.

**Comment by faern** [comment]
If anyone can traffic dump this bookmark synchronization with `tcpdump` or Wireshark or similar and help us figure out what kind of traffic it is that's needed to allow this, we can probably fix it way faster.

It's probably using some kind of local *casting-somethingsomething address/port combination that is currently blocked in the firewall. If we figure out what and then determine that allowing it is not against our security policies, then we can just unblock it in the firewall.

**Comment by paulrudy** [comment]
I've done a Wireshark capture just now. Is there something I should look or filter for?

**Comment by pinkisemils** [comment]
I did one too, and, when excluding `17.0.0.0/8` (apple's subnet), I saw no traffic. Probably an error on my end. Is there a deterministic way of making the synchronization take place? Does any iCloud syncing work when the app is connected?

**Comment by paulrudy** [comment]
@faern Please advise what to filter for in a Wireshark dump, I'm happy to share what I find but unfamiliar with Wireshark.

**Comment by faern** [comment]
@paulrudy Anything going to some local multicast address. Might even be IPv6 for all I know. I'm not sure what to filter for. But if you try to exit all other programs so that the computer is not so chatty and repeat the experiment a few times. Maybe you'll see some packets going out or coming in around the time when the sync happens that are similar every time the sync happens?

**Comment by paulrudy** [comment]
@faern I ran three tests, one after the next, where I started Wireshark and immediately either created or deleted a Safari bookmark. In each test there's a lot of chat between my local ip and 17.248.188.xxx (the last 3 digits were different in each test). Looking those ip's up show that they belong to Apple. Is that useful or do I need to dig deeper?

**Comment by faern** [comment]
Yes, the entire `17.x.x.x` net is Apple's. But that has nothing to do with the LAN, that's on the internet. Our VPN app does not prevent communication with that IP range. So if their bookmark sync is not performed locally, but rather via their internet servers, then I don't see how we would be blocking it.

**Comment by paulrudy** [comment]
I don't know if you saw this [earlier comment](https://github.com/mullvad/mullvadvpn-app/issues/2401#issuecomment-984302199), but I've also confirmed that connecting to Mullvad tunnels via the WireGuard official app does *not* interfere with iCloud Safari bookmark sync. So it's something about the Mullvad app. Disabling "black ads" and "block trackers" does not seem to make a difference.

In case it's useful, the IVPN app also breaks iCloud Safari bookmark sync

**Comment by possiblerobot** [comment]
This might help. When trying to sync bookmarks over Mullvad, the console shows this message:

```
502:com.apple.SafariBookmarksSyncAgent.XPC.BookmarkSyncNetworkConnectivity:2A144B:[
	{name: NetworkQualityPolicy, policyWeight: 8.400, response: {Decision: Must Not Proceed, Score: 0.00, Rationale: [{[wiredQuality]: Required:20.00, Observed:0.00},{[wifiQuality]: Required:50.00, Observed:0.00},{[networkPathAvailability]: Required:1.00, Observed:1.00},]}}
 ], FinalDecision: Must Not Proceed}
```

When the VPN is disconnected, you get this:

```
502:com.apple.SafariBookmarksSyncAgent.XPC.BookmarkSyncNetworkConnectivity:FB9C66:[
	{name: DeviceActivityPolicy, policyWeight: 2.000, response: {Decision: Can Proceed, Score: 0.65}}
 ] sumScores:38.210000, denominator:38.910000, FinalDecision: Can Proceed FinalScore: 0.982010}
```

Could this issue have something to do with this NetworkQualityPolicy / DeviceActivityPolicy stuff?

**Comment by pinkisemils** [comment]
Seems like this will require the app to use Apple's [VPN API](https://developer.apple.com/documentation/networkextension/packet_tunnel_provider) instead of just using unixy APIs to create a tunnel device to circumvent these issues. Or maybe there's a better way to inform the system that the routes we've added are legitimate and do work.

**Comment by paulrudy** [comment]
@pinkisemils Does Wireguard's app use Apple's VPN API? Because, as mentioned, these problems don't occur when connecting through the Wireguard app (to mullvad servers)

**Comment by jgogstad** [comment]
Just confirmed that it's the macOS side that's broken:

1. iPhone with Mullvad 2021.4 running
2. macOS with 2021.6

Bookmarks syncs once I disable mullvad on the macOS side.

For what it's worth bookmarks should be synced with https according to https://support.apple.com/en-gb/HT202944. I've done a couple of captures while seeing a bookmark being synced to my phone, and I only see traffic on TCP 443.

**Comment by paulrudy** [comment]
Is there anything else we can provide that will help solve this bug? Just reminding the devs also that the Wireguard app does not share this bug, it's specific to the macOS version of the Mullvad app.

**Comment by paulrudy** [comment]
I haven't checked in a few weeks, but as of today:
- Safari tab groups sync on macOS with VPN connected via Mullvad app.
- Safari bookmarks do not sync on macOS with Mullvad VPN. Previously this only occurred when using the Mullvad app, but not the Wireguard app. Now it occurs with both apps
- Safari bookmarks do sync when not connected to any VPN

**Comment by n-stone** [comment]
iCloud (File) sync is broken for me as well, if I add a file on my iCloud drive on the iPhone it won't appear until I turned off the Mullvad App on my Mac.

I'm on macOS 12.2.1 with Mullvad 2022.1-beta.

**Comment by dahmc** [comment]
@n-stone Works fine for me with Desktop 2021.6 and iOS 2022.1
Looks like a very bad regression then? Anyone else have the issue?

**Comment by n-stone** [comment]
I tested it again, it seems to be something sleep related on the macOS site. If the mac is awake und "fresh" connected via the mullvad App everything is fine. But if the mac is connected via mullvad goes in some (deeper??? 5 Minutes is not enough) sleep state and wakes up connects again via mullvad App the iCloud Drive sync won't work.
Tricky to test / reproduce.

**Comment by steve10883** [comment]
> I haven't checked in a few weeks, but as of today:
> 
> * Safari tab groups sync on macOS with VPN connected via Mullvad app or Wireguard app.
> * Safari bookmarks do not sync on macOS with Mullvad VPN connected via Mullvad app or Wireguard app. Previously this only occurred when using the Mullvad app, but not the Wireguard app. Now it occurs with both apps.
> * Safari bookmarks do sync when not connected to any VPN.

I am not seeing this behaviour at all. Using the Wireguard app 1.0.15, Safari 15.3 and macOS 12.2.1. Bookmarks and Reading List sync without any issue.

**Comment by n-stone** [comment]
> I tested it again, it seems to be something sleep related on the macOS site. If the mac is awake und "fresh" connected via the mullvad App everything is fine. But if the mac is connected via mullvad goes in some (deeper??? 5 Minutes is not enough) sleep state and wakes up connects again via mullvad App the iCloud Drive sync won't work.
> Tricky to test / reproduce.

This is what it looks like:
![image](https://user-images.githubusercontent.com/31704359/156234612-ca6e5d39-f1e0-47fa-b6c4-f1b43c136cfa.png)

After I woke up the mac I straight up created a new folder with a File in it. iCloud Drive tries to upload it and is stuck at around 90%. It will stay like this until I disconnect Mullvad. The new File will not appear in the iCloud drive on my iPhone until the upload is finished. I waited around 10 Minutes before I disconnected Mullvad, Mullvad was fully connected and am.i.mullvad was working and showing the correct server.

**Comment by mcmurry-1** [comment]
I'm jumping on here to say that I also have this issue.

On macOS, an active VPN connection with the Mullvad app breaks basic system features, like bookmark syncing. I also often see the iCloud Drive issue mentioned above by @n-stone. 

The bookmarks start syncing a few seconds after disconnecting Mullvad, by initiating a connection to gateway.icloud.com. With an active Mullvad connection, this domain just stays silent, the bookmark changes aren't seen.

I think it's very underwhelming that broken system functionality persists for more than 1 year after this was first mentioned here.

**Comment by steve10883** [comment]
> > I tested it again, it seems to be something sleep related on the macOS site. If the mac is awake und "fresh" connected via the mullvad App everything is fine. But if the mac is connected via mullvad goes in some (deeper??? 5 Minutes is not enough) sleep state and wakes up connects again via mullvad App the iCloud Drive sync won't work.
> > Tricky to test / reproduce.
> 
> This is what it looks like: ![image](https://user-images.githubusercontent.com/31704359/156234612-ca6e5d39-f1e0-47fa-b6c4-f1b43c136cfa.png)
> 
> After I woke up the mac I straight up created a new folder with a File in it. iCloud Drive tries to upload it and is stuck at around 90%. It will stay like this until I disconnect Mullvad. The new File will not appear in the iCloud drive on my iPhone until the upload is finished. I waited around 10 Minutes before I disconnected Mullvad, Mullvad was fully connected and am.i.mullvad was working and showing the correct server.

Some sleep related issues have been fixed in the latest release https://mullvad.net/en/blog/2022/3/1/macos-now-reconnects-instantly-after-being-sleep-mode/

Unfortunately still no fix for bookmark and reading list syncing. 14 months and counting.

**Comment by pronebird** [comment]
iCloud Drive seems to work fine for me when on VPN. macOS 11.6.4.

**Comment by paulrudy** [comment]
FYI, with Mullvad 2022.1, Safari tab groups sync and Safari bookmark sync continue to be broken.

**Comment by n-stone** [comment]
@steve10883 I was on the beta before and now updated to 2022.1, error persists.
@pronebird The upload error only happens when the mac was put to sleep with mullvad on and directly connects to mullvad after sleep. If I turn of Mullvad put it sleep and connect mullvad again after sleeping everything is fine.

**Comment by faern** [comment]
Thanks for all the extra information provided here. The iCloud sync issue is probably easier for us to test. So weird that it will sync to 90% and then give up? I would assume it would not sync at all or finish.

I have created a new internal issue for this in particular. Maybe looking at iCloud drive will shine some light on what's up here.

**Comment by notDavid** [comment]
Fyi, i just noticed `kill bird` (this restarts the back-end process behind iCloud) solves the issue for me, so it's a quick workaround to initiate an iCloud sync without rebooting or whatever. Perhaps this works for other too...

Mullvad 2022.1
macOS 12.2.1

**Comment by mcmurry-1** [comment]
> Fyi, i just noticed `kill bird` in Terminal (this restarts the back-end process behind iCloud) solves the issue for me, so it's a quick workaround to initiate an iCloud sync without rebooting or whatever. Perhaps this works for other too...
> 
> Mullvad 2022.1 macOS 12.2.1

I'm assuming you are talking about iCloud Drive syncing here, because the bookmark sync issue is not affected by this process at all.

**Comment by notDavid** [comment]
> because the bookmark sync issue is not affected by this process at all and still unsolved

@mcmurry-1 Ah, yes indeed.

Out of curiosity, you could perhaps try: `launchctl kickstart -k gui/501/com.apple.SafariBookmarksSyncAgent`
I think this should restart the bookmark sync service.

**Comment by paulrudy** [comment]
> Out of curiosity, you could perhaps try: `kickstart -k gui/$(id -u)/com.apple.SafariBookmarksSyncAgent` I think this should restart the bookmark sync service.

@notDavid I tried to test this, but got `zsh: command not found: kickstart`

**Comment by notDavid** [comment]
@paulrudy Sorry, correction:

`launchctl kickstart -k gui/$(id -u)/com.apple.SafariBookmarksSyncAgent`

**Comment by paulrudy** [comment]
@notDavid thanks. Unfortunately, that command didn't help. After entering it, I continued to get log items like this, and bookmarks don't sync:

```
501:com.apple.SafariBookmarksSyncAgent.XPC.CloudTabGroupZoneSubscriptionRegistration:D1E2AB:[
	{name: NetworkQualityPolicy, policyWeight: 8.400, response: {Decision: Must Not Proceed, Score: 0.00, Rationale: [{[wiredQuality]: Required:20.00, Observed:0.00},{[wifiQuality]: Required:50.00, Observed:0.00},{[networkPathAvailability]: Required:1.00, Observed:1.00},]}}
 ], FinalDecision: Must Not Proceed}
```

Once mullvad was disconnected, bookmarks synced fine.

**Comment by mcmurry-1** [comment]
> @paulrudy Sorry, correction:
> 
> `launchctl kickstart -k gui/$(id -u)/com.apple.SafariBookmarksSyncAgent`

This only works to start a sync when not connected to Mullvad. With a Mullvad VPN connection, it does nothing. The process responsible for syncing bookmarks (cloudd or nsurlsessiond) does not initiate a connection to gateway.icloud.com.

**Comment by ElysiaAtroviridis** [comment]
> FYI, with Mullvad 2022.1, Safari tab groups sync and Safari bookmark sync continue to be broken.

Adding iCloud Tabs to the list of broken items while the MacOS App is active, at least on my end.

**Comment by paulrudy** [comment]
> > FYI, with Mullvad 2022.1, Safari tab groups sync and Safari bookmark sync continue to be broken.
> 
> Adding iCloud Tabs to the list of broken items while the MacOS App is active, at least on my end.

Confirming that iCloud Tabs is broken with MacOS Mullvad app active

**Comment by ElysiaAtroviridis** [comment]
Dear commentators, how have you've been working around the issue? In other to prevent Safari from breaking after turning on the VPN I've done all of these, to **no avail**: 

- Stop using the iOS Mullvad App entirely (which is an option I have but would ideally not need to resign to)
- Stop using extensions either on MacOS or iOS
- Use the Wireguard App instead of Mullvad's own
- Stop/Switch using Apple Private Relay either on MacOS or iOS
- ...

Note that turning the VPN (either via Mvad or WGuard App) breaks Safari sync for me until manually resetting the app entirely using silly CleanMyMac'S reset feature for apps that are locked from full deletion. Plus rebooting.

I'm not sure why I haven't simply switched browsers across devices at this point…

**Comment by mcmurry-1** [comment]
> Dear commentators, how have you've been working around the issue? In other to prevent MacOS Safari from breaking after turning on the VPN I've done all of these, to **no avail**:
> 
> * Stopped using the iOS Mullvad App entirely (which is an option I have but would ideally not need to resign to)
> * Stopped using Safari extensions either on MacOS or iOS (¯_(ツ)_/¯)
> * Use the Wireguard App instead of Mullvad's own on MacOS
> * Stop/Switch using Apple Private Relay either on MacOS or iOS
> * ...
> 
> Note that turning the VPN (either via Mvad or WGuard App) breaks Safari sync for me until manually resetting the app entirely using silly CleanMyMac's reset feature for apps that are locked from full deletion. Plus rebooting.
> 
> I'm not sure why I haven't simply switched browsers across devices at this point…

Temporarily disconnecting Mullvad VPN on Mac works to sync bookmarks. Making a change to bookmarks (renaming, moving, adding, deleting) will kickstart the process and sync to other devices within 20 seconds in my experience.

There is no need to stop using the iOS Mullvad app because only the macOS Mullvad app breaks this functionality. 
As long as the Mac app is disconnected for a while, I can see changes syncing from macOS to iOS and from iOS to macOS without issues, while still being connected to Mullvad on iOS (as others here have confirmed as well).

This issue has nothing to do with browser extensions, Apple Private Relay, resetting app preferences or even Safari itself. The process that starts bookmark syncing just is not initiated with an active Mullvad connection on macOS. That's all. The only way to fix it is to wait for Mullvad to fix the bug, or stop using Mullvad.

**Comment by ElysiaAtroviridis** [comment]
> > Dear commentators, how have you've been working around the issue? In other to prevent MacOS Safari from breaking after turning on the VPN I've done all of these, to **no avail**:
> > 
> > * Stopped using the iOS Mullvad App entirely (which is an option I have but would ideally not need to resign to)
> > * Stopped using Safari extensions either on MacOS or iOS (¯_(ツ)_/¯)
> > * Use the Wireguard App instead of Mullvad's own on MacOS
> > * Stop/Switch using Apple Private Relay either on MacOS or iOS
> > * ...
> > 
> > Note that turning the VPN (either via Mvad or WGuard App) breaks Safari sync for me until manually resetting the app entirely using silly CleanMyMac's reset feature for apps that are locked from full deletion. Plus rebooting.
> > I'm not sure why I haven't simply switched browsers across devices at this point…
> 
> Temporarily disconnecting Mullvad VPN on Mac works to sync bookmarks. Making a change to bookmarks (renaming, moving, adding, deleting) will kickstart the process and sync to other devices within 20 seconds in my experience.
> 
> There is no need to stop using the iOS Mullvad app because only the macOS Mullvad app breaks this functionality. As long as the Mac app is disconnected for a while, I can see changes syncing from macOS to iOS and from iOS to macOS without issues, while still being connected to Mullvad on iOS (as others here have confirmed as well).
> 
> This issue has nothing to do with browser extensions, Apple Private Relay, resetting app preferences or even Safari itself. The process that starts bookmark syncing just is not initiated with an active Mullvad connection on macOS. That's all. The only way to fix it is to wait for Mullvad to fix the bug or stop using Mullvad (temporarily).

Thank you @mcmurry-1. My list came about since temporarily disconnecting Mullvad VPN does not work on my Mac to restart the sync process, at least for me. Neither did the terminal commands other users have kindly shared. 

To read that the iOS app does not affect this issue once more is helpful however. I had jumped the ship in attempt to minimize possible error sources…

**Comment by mcmurry-1** [comment]
> My list came about since temporarily disconnecting Mullvad VPN does not work on my Mac to restart the sync process, at least for me. Neither did the terminal commands other users have kindly shared.

@ElysiaAtroviridis I see, that's weird. Does it work when you bookmark something or edit an existing bookmark (like changing the name, moving, …)? Whenever I do this (with macOS Mullvad disconnected), I can see the changes "live" on iOS within a few seconds.

**Comment by ElysiaAtroviridis** [comment]
Sadly no. Changes remain *stuck* on individual devices until I start a choreography of unsuccessfully toggling Safari iCloud sync and eventually resolve to resetting the Mac app entirely.

**Comment by mcmurry-1** [comment]
> Sadly no. Changes remain _stuck_ on individual devices until I start a choreography of unsuccessfully toggling Safari iCloud sync and eventually resolve to resetting the Mac app entirely.

Are you sure this is related to Mullvad in your case though? 

If you quit Mullvad, remove it from your login items, reboot our Mac (without starting Mullvad) and it still doesn't work, there might be something else at play. Some aspects of iCloud sync data don't ever work for me either, like iCloud tabs. I can't even close them manually, they are just stuck in a random state from 2 years ago.

**Comment by ElysiaAtroviridis** [comment]
It’s a slim margin, but what kept me coming back here is the fact that it all works flawlessly after resetting. Until the very moment I turn on Mullvad/Wireguard on MacOS once.

**Comment by mcmurry-1** [comment]
> It’s a slim margin, but what made me come back here is the fact that it all works flawlessly after resetting. Until the very moment I turn on Mullvad/Wireguard on MacOS once.

Ok, that sounds like it's related. How annoying.

One similar situation can be seen with Private Relay, in that starting a macOS Mullvad connection breaks the feature until you restart your Mac. Even after disconnecting Mullvad, you can not turn on Private Relay in the settings. It's completely broken until you restart the Mac (without Mullvad autoconnecting).

**Comment by steve10883** [comment]
> > The only way to fix it is to wait for Mullvad to fix the bug or stop using Mullvad (temporarily) on macOS.

At this point in time it’s safe to assume it’s never going to be done. The issue has been ongoing for 15 months with no progress at all.

If you’re using macOS I’d recommend you jump ship to another provider that actually gives a crap about their users.

**Comment by dionvl** [comment]
@ElysiaAtroviridis I have switched over to iCloud Private Relay, I haven't had a subscription for months now. I'd still prefer to use Mullvad for full protection in all apps, which is why I'm still following this thread after all this time. My primary concern is DNS + browser (Safari) + unsecured traffic for public wifi. Private Relay does cover all that so it's fine for now I suppose.

**Comment by aaronraimist** [comment]
I'm guessing this must be caused by some API that the Mullvad app doesn't implement. When I use [the WireGuard app](https://apps.apple.com/us/app/wireguard/id1451685025) with Mullvad config files everything syncs normally.

When I use the Mullvad app nothing related to iCloud syncs. In Console.app `dasd` continually logs things like:

```
com.apple.CFNetwork-cc-614-1772-Task <75209A5E-AC66-41A1-95F8-51088246EA6D>.<143>:A70229:[
	{name: NetworkQualityPolicy, policyWeight: 11.400, response: {Decision: Must Not Proceed, Score: 0.00, Rationale: [{[wiredQuality]: Required:20.00, Observed:0.00},{[wifiQuality]: Required:100.00, Observed:0.00},{[networkPathAvailability]: Required:1.00, Observed:1.00},]}}
 ], FinalDecision: Must Not Proceed}
```

so macOS seems to think the network quality is 0 and therefore doesn't attempt to sync.

**Comment by ElysiaAtroviridis** [comment]
@aaronraimist do have said Wireguard configuration running permanently? (And are you using the Mullvad iOS App (toggled on) alongside it?) Sadly using Wireguard on MacOS has broken Safari for me the last time around, after initially thinking it had done the trick.

**Comment by ElysiaAtroviridis** [comment]
@notDavid I can attest to all things sync repeatedly breaking after/during active Mullvad connections using the app on MacOS. 

My workaround after way too much time wasted has become using Wireguard and LittleSnitch, counting on remembering/LN  recognizing if I've got a VPN connection active and switching (and remaining in) a custom connection profile. This is not a real kill switch to my understanding, but as close as I came feeling reasonable within the given effort.

**Comment by aaronraimist** [comment]
@notDavid I don't believe it is bypassing the Wireguard VPN in this case.

I just did an experiment where I uploaded a video to my photos library while not doing much else on my computer. While it was uploading I looked at the network page of iStat Menus. The WireGuardNetworkExtension upload rate closely matched (and always exceeded) nsurlsessiond for the entire upload. When the upload stopped, both reduced to around 0 KB/s at the same time.
<img width="313" src="https://user-images.githubusercontent.com/5855073/166264101-aa549737-20e7-439d-829e-071fc1fb5ab1.png">
<img width="313" src="https://user-images.githubusercontent.com/5855073/166264146-2ed8df6d-07a4-43f1-8582-44ee8c32250a.png">
<img width="309" src="https://user-images.githubusercontent.com/5855073/166264122-c5ebc84e-bf38-49e7-80ff-61fbcd207dfc.png">

At the end of the upload, the network page of Activity Monitor show that the amount uploaded by WireGuardNetworkExtension slightly exceeded the amount uploaded by nsurlsessiond.

<img width="1177" src="https://user-images.githubusercontent.com/5855073/166264493-63649a14-f6f6-4918-8f0f-ef2662f00771.png">

I have observed this same result many times. You should be able to reproduce this yourself. If someone can give me a `tcpdump` command or something else that would confirm it is using the VPN, I would be happy to try it.

**Comment by possiblerobot** [comment]
I would really love a resolution on this syncing issue. It's the single biggest drawback to using Mullvad right now. Otherwise, it's an amazing service! I haven't switched away yet, but this problem gives me a wandering eye.

**Comment by n-stone** [comment]
I guess you know this support document: https://support.apple.com/en-us/HT210060

# iCloud
In addition to the Apple ID hosts listed above, Apple devices must be able to connect to hosts in the following domains to use iCloud services.


Hosts | Ports | Protocol | OS | Description | Supports proxies
-- | -- | -- | -- | -- | --
*.apple-cloudkit.com | 443 | TCP | iOS, iPadOS, tvOS, and macOS | iCloud services | —
*.apple-livephotoskit.com | 443 | TCP | iOS, iPadOS, tvOS, and macOS | iCloud services | —
*.apzones.com | 443 | TCP | iOS, iPadOS, tvOS, and macOS | iCloud services in China | —
*.cdn-apple.com | 443 | TCP | iOS, iPadOS, tvOS, and macOS | iCloud services | —
*.gc.apple.com | 443 | TCP | iOS, iPadOS, tvOS, and macOS | iCloud services | —
*.icloud.com | 443 | TCP | iOS, iPadOS, tvOS, and macOS | iCloud services | —
*.icloud.com.cn | 443 | TCP | iOS, iPadOS, tvOS, and macOS | iCloud services in China | —
*.icloud.apple.com | 443 | TCP | iOS, iPadOS, tvOS, and macOS | iCloud services | —
*.icloud-content.com | 443 | TCP | iOS, iPadOS, tvOS, and macOS | iCloud services | —
*.iwork.apple.com | 443 | TCP | iOS, iPadOS, tvOS, and macOS | iWork documents | —
mask.icloud.com | 443 | UDP | iOS, iPadOS, macOS | iCloud Private Relay | —
mask-h2.icloud.com | 443 | TCP | iOS, iPadOS, macOS | iCloud Private Relay | —
mask-api.icloud.com | 443 | TCP | iOS, iPadOS, macOS | iCloud Private Relay | Yes

Would be great if the mullvad app had a Whitelisting feature.

**Comment by paulrudy** [comment]
In case it's useful, the ProtonVPN does not cause issues for me with Safari Bookmarks sync over iCloud.

**Comment by paulrudy** [comment]
@notDavid Oh I see. Yes, when I enable ProtonVPN's kill switch, Safari Bookmarks sync stops working.

I would be grateful if there could be some option for a resolution to this issue in Mullvad, perhaps as @n-stone said, a whitelisting option.

**Comment by paulrudy** [comment]
@faern, is there any progress on this issue? Would whitelisting the relevant processes be the solution, and if so can that be done?

**Comment by pallorc** [comment]
I just tested Safari bookmark syncing with the latest Mullvad beta (2022.5 beta-1) and syncing still doesn't work. Any word on whether this issue is getting worked on? - Thanks

**Comment by 75231** [comment]
Preface: I don't have much technical background here. I'm posting this to hopefully provide additional context, not just to kvetch. I know that maintaining libraries is difficult, and issues are often more complicated than they appear. Also, I love Mullvad.

All of that said, this is a major issue for Apple users, who represent (I assume) a significant number of Mullvad users.

1. Apple users are probably more likely than others to be all-in on the "Apple ecosystem", meaning they have multiple devices.
2. iCloud is the main way that these devices are kept in sync.
3. iCloud is (to my admittedly-untrained eye) pretty naive about syncing some things. e.g. If I bookmark something on my Mac and I bookmark something on my iPhone before it's had a chance to pull down the latest changes, the bookmarks on my iPhone are treated as a block and they overwrite the ones on my Mac. This means that any connection issue between an Apple device and the iCloud server is very likely to lead to data loss. I haven't tested whether iCloud Drive has the same problem, but if it does that's far more potentially problematic than Safari.
4. Mullvad is likely contributing to a diminished user experience and even data loss for a significant number of its users.

I (and probably a lot of people) have had this page bookmarked for almost two years and I'm somewhat confused and disheartened that this issue hasn't been addressed. I love Mullvad and have been a user for years. I've set multiple friends and family members up on Mullvad. This issue, however, has made me second guess that decision. This is a major problem impacting key features of multiple major platforms and it seems like it should've been prioritized sometime in the last 2 years.

If Mullvad has decided that this level of support is appropriate for these platforms, please let us know. I know tradeoffs have to be made sometimes. If that's a tradeoff that the organization has consciously decided to make, I'll sadly have to seek out another VPN. I just want to be able to make an informed decision here.

**Comment by 75231** [comment]
Adding to the list of iCloud issues, Mullvad's Mac app also appears to break upward reporting of Screen Time.

**Comment by pinkisemils** [comment]
A viable solution that allows us to preserve the current privacy guarantees has been found, and we'll work on it _very soon_. 

The solution is painfully simple - the primary interface's default route must be changed to be scoped to the interface and a default route through the tunnel interface must be added. Turns out, Apple's frameworks don't really use the routing table the same way normal applications do - only the highest priority default route is used, even if more specific routes for the destination exist in the routing table.

**Comment by paulrudy** [comment]
> A viable solution that allows us to preserve the current privacy guarantees has been found, and we'll work on it _very soon_. 
> 
> 
> 
> The solution is painfully simple - the primary interface's default route must be changed to be scoped to the interface and a default route through the tunnel interface must be added. Turns out, Apple's frameworks don't really use the routing table the same way normal applications do - only the highest priority default route is used, even if more specific routes for the destination exist in the routing table.

Very happy to hear this. Thank you for looking into it!

**Comment by 75231** [comment]
After updating to Ventura, FaceTime no longer works with Mullvad enabled. Is this problem related or should I file a separate issue?

**Comment by nihaals** [comment]
> After updating my Mac to Ventura, FaceTime no longer works with Mullvad enabled. It still works on iOS.

To add more context (at least based on my experience) for people who find this issue:
- Making a FaceTime call just keepings ringing without the recipient actually receiving a call and with no indication of the call failing
- FaceTime calls aren't received
- FaceTime Handoff doesn't show up
- If connecting in the middle of a call, it immediately freezes, and immediately unfreezes after disconnecting
- Same situation described above where using the Mullvad client fails and using the WireGuard Mac App Store client works fine
- Support said it's a known issue

macOS 13.0.1, Mullvad 2022.5

**Comment by 75231** [comment]
Everything @nihaals says reflects my experience too.

I probably won't post any more updates here. I'm going to start looking for another VPN provider and I won't be recommending Mullvad to Mac users in the future. I won't pretend to know why this is the case, but it's clear that Mullvad can't or won't allocate the resources necessary to support the basic functionality of Mac OS.

I am disappointed that my preferred OS isn't prioritized, but you can't cater to all use cases. Thanks to Mullvad for all of the important work they do.

**Comment by paulrudy** [comment]
@75231 Please let us know if you find reputable alternatives. I'm disappointed that Mullvad isn't implementing a solution to this issue, even when @pinkisemils suggests the solution is "painfully simple".

**Comment by steve10883** [comment]
@notDavid I disagree.

1. Who is going to decide which comments are useless? You? A number are people's experiences and trying to provide Mullvad with information in an attempt to get it fixed, because they don't show much interest in doing so.
2. If you're finding the 'deluge' (/s) of notifications too much than you're probably a bit sensitive for the internet.
3. Mullvad don't have a message forum. Users come here in an attempt to find out what's going on.
4. Posting lets Mullvad know users do care and they want it fixed, and that there are enough macOS users out there for them to expend resources on the issue.
5. By hiding comments it would look suspiciously like Mullvad are trying to bury bad news.

And let's face it. It's almost two years since this was reported, and not having fixed it yet is just outright negligence on Mullvad's part.

**Comment by pinkisemils** [comment]
@paulrudy Mullvad is implementing the solution. I am implementing the solution. 
It's painfully simple in concept, but it requires listening to changes to the routing table, and this isn't something that is well documented. I do apologize for not working faster on it.

**Comment by paulrudy** [comment]
> @paulrudy Mullvad is implementing the solution. I am implementing the solution. 

I'm glad to hear it, thank you

> It's painfully simple in concept, but it requires listening to changes to the routing table, and this isn't something that is well documented. I do apologize for not working faster on it. 

I appreciate the transparency, @pinkisemils thank you!

**Comment by eskawater** [comment]
is it just me or iCloud Drive seems to be working since ventura 13.2? Before while connected to mullvad if I created a folder in icloud drive on macos it would not sync to the iPhone. Seems to be working fine now.

**Comment by paulrudy** [comment]
> is it just me or iCloud Drive seems to be working since ventura 13.2? Before while connected to mullvad if I created a folder in icloud drive on macos it would not sync to the iPhone. Seems to be working fine now.

I'm not currently subscribed to mullvad to test, but could you check whether Safari tab and Safari bookmark syncing seems to work as well? If so, I'll gladly switch back to mullvad

**Comment by eskawater** [comment]
unfortunaly safari bookmarks is still broken

**Comment by pinkisemils** [comment]
It is still broken, and I am still working on it. It is in the later stages where I have a version of the app that works as intended in the nominal case, but I need to rework how the app handles changes in network adapters (connecting to ethernet after already being connected over WiFi). Unfortunately, this is no small change. I can promise that as soon as the changes are ready for review, this issue will be updated. I will also probably post a link to a version you can try out before an official beta is released just to see what else my changes could break. 

Having said all of that, I and the app team are fairly well informed of the current state of this particular issue. Whilst we appreciate the pro-active engagement from our users about this and any other issue they might have with our app, in this particular issue, we don't believe restating the obvious (the fact that the app is still broken in this way) is helpful for anyone.

**Comment by pallorc** [comment]
Thanks @pinkisemils for working on this issue and finding a solution. The effort you put in is appreciated!

**Comment by RhvdG** [comment]
We are still waiting for the fix. Is it forthcoming?

**Comment by pinkisemils** [comment]
I am still working on it, it's inching closer by the day. Unfortunately, it's probably not going to be released in the upcoming release. But I will post a build here as soon as I've verified that it works and doesn't set anything on fire. I do apologize for the slow progress on this issue, but, as I've said before, the changes aren't exactly trivial.

**Comment by Drakula2k** [comment]
@notDavid Home still works fine for me on 13.3, make sure you allow access to your local network with "VPN Settings" -> "Local network sharing"

**Comment by santisbon** [comment]
Not sure if there are plans for a Mullvad app on the **macOS** App Store but for other VPN providers the app downloaded from the company's website blocks iCloud Drive sync and the version from the App Store doesn't. So maybe the App Store approval process for macOS apps would be a good place to look for exactly what causes the problem.

**Comment by acrostich** [comment]
I was wondering whether to post this as I haven't tested it thoroughly: I have noticed that in situation when iCloud Drive is stuck on Upload x items in Finder I'm using a single hop, when I'm using multihop (both on Wireguard) than iCloud Drive tends to sync. It doesn't make much sense and may be related to something else when I noticed this, but posting anyway it in case it may help.

**Comment by dlon** [comment]
We've had more time to work on this recently and finally have a test build: https://releases.mullvad.net/builds/2023.4-beta1-dev-5a37e3+test-macos-routing-rework/MullvadVPN-2023.4-beta1-dev-5a37e3+test-macos-routing-rework.pkg

Please try it out and report any issues that you find.

**Comment by dlon** [comment]
> @dlon Thank you. So what does this do, exclude specific Apple iCloud IP addresses from the firewall & vpn routing table?

@pinkisemils's [comment](https://github.com/mullvad/mullvadvpn-app/issues/2401#issuecomment-1283821590) explained it pretty well. We're just keeping track of the best interface/route to use for sending traffic to the relay/server, and making sure that the tunnel interface always has a default route to placate some of Apple's frameworks.

**Comment by ddoggerel** [comment]
> Please try it out and report any issues that you find.

I've tested it, and it has resolved all of the issues that had forced me to stop using the Mullvad app. Thank you very much.

The issues that I had, and that are now fixed:

1. With Private Relay enabled, making a Mullvad connection resulted in PR being disabled (expected behaviour), BUT after Mullvad connection was terminated, PR would remain disabled until a complete reboot of the system. This is now fixed so that PR returns to an enabled state when the Mullvad VPN connection is closed. This issue had been present until superseded by the following issue sometime last Fall.

2. Sometime last Fall, having the Mullvad app installed (even when there is no active VPN connection) prevented Private Relay from being active. This is now fixed.

3. I never directly observed the interference with iCloud syncing, but can confirm that on my system this Mullvad beta is not preventing iCloud syncing.

Current system is Ventura 13.3.1 running on a MBP M1 Pro.
The Mullvad beta that I tested is: "MullvadVPN-2023.4-beta1-dev-5a37e3+test-macos-routing-rework".

Thanks again!

**Comment by pallorc** [comment]
Just installed and tested Safari bookmark syncing. It works! 🙌  I'll post more if I discover any issues or notable behaviors. Any expectations on when this change will be rolled into the mainline app (barring finding a significant issue)?

Thank you for finding a solution!

**Comment by philml** [comment]
Excellent! Looks all good here -- bookmarks, calendar, notes and iCloud files on Desktop are all syncing both ways from mac to iPad, even when the Mac is connected to Mullvad!

**Comment by Bandolorian** [comment]
Also like to report that iCloud file syncing is working with the beta build above on the Mac.

**Comment by dlon** [comment]
@acrostich Thanks, fixed!

I'm closing this thread as we merged the fix for this a couple of days ago. You can probably expect it to be part of the 2023.5 release (not the next stable, but the stable version after that).
